staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -199,7 +199,7 @@ func (w *WebhookAuthorizer) Authorize(ctx context.Context, attr authorizer.Attri
 		}
 		r.Status = result.Status
 		if shouldCache(attr) {
-			if r.Status.Allowed {
+			if r.Status.Allowed && !r.Status.Denied {
```

Comment:
Judging by how line 210 below treats this case, this change is correct.

However, this correctness change should come with a test.

The consequence is very small and we are possibly going to change this caching behavior anyway, so it might not be worth it, though. @mikedanese @liggitt

Reply:
Even though this code may change in the future, I think it is good practice to maintain consistent code.

Let me see if I can come up with a test.

Reply:
authorizer.DecisionDeny and authorizer.DecisionAllow are not bitmasks.
This makes constructing Authorizer which has both set in decision a bit difficult.

Reply:
At the beginning of WebhookAuthorizer#Authorize, we construct key and use it to read from w.responseCache .
One test I am thinking of is to construct key with the same code and put v1beta1.SubjectAccessReviewStatus object inside w.responseCache using the key.
This way, the test has control over both Allowed and Denied fields of the status.

@lavalamp 
Can you confirm whether the plan above is okay or not okay ?

Reply:
I thought about injecting an entry into w.responseCache but that bypasses the else block where my change takes place.

I will keep thinking about how to add test for my change.